### DISCUSSION TITLE
Watch for contract changes to run deploy/test

### DISCUSSION THREAD

**Discussion by ByteAtATime** [scaffold-eth/scaffold-eth-2#865] [Ideas]
I believe a nice QoL feature would be to have a command to automatically deploy/test contracts when they change. I'm definitely down to submit a PR for this, but I'm thinking we use [`nodemon`](https://www.npmjs.com/package/nodemon) for watching/command running. I would definitely consider this a "minor change", but just wanted to open a discussion for whether this is needed and if my approach is valid.

I was thinking we add two extra commands, each one listening to changes in the `contracts` and `deploy` directories:
1. `yarn deploy:watch`, which runs `yarn deploy` every time the contracts/deploy scripts change
2. `yarn hardhat:test:watch` or `yarn hardhat:test-watch`, which runs `yarn hardhat:test` every time any of the contracts, deploy scripts, or tests change

Regarding the second command, I'm not sure if I'm a fan of the triple colons, but I'm not sure what the best way to deal with this would be. Again, whatever method we end up with, I'll create a PR!

**Comment by carletex** [comment]
`yarn deploy --watch` is already doing (1).

I think (2) is not worth it, especially if we need to add an extra dep.
